### ISSUE TITLE
After Alt+Tab back into Mintty, sometimes keys behave as if Ctrl is continuously pressed.

### ISSUE THREAD

**Issue by DoctorNoobingstoneIPresume** [mintty/mintty#1273]
If in _Options -> Window -> Modifier for scrolling_ I select _Ctrl_, then I Alt+Tab in another application, and then I Alt+Tab back in Mintty, the keyboard sometimes behaves as if Ctrl is being pressed together with the keys I actually press.  :'(

To escape from this behaviour, I have to press both Ctrl keys of my keyboard.

**Comment by mintty** [comment]
So pressing e.g. a letter 'c' then enters a control+C? I don't reproduce that.
Does it also happen if the other application is another mintty window?
What do you mean by "sometimes"?
Do you have any desktop software (like hotkeys) installed?

**Comment by DoctorNoobingstoneIPresume** [comment]
> So pressing e.g. a letter 'c' then enters a control+C? I don't reproduce that. Does it also happen if the other application is another mintty window? What do you mean by "sometimes"? Do you have any desktop software (like hotkeys) installed?

Thank you. Partial answer I can currently give:

- Yes, pressing 'c' results in `^C` being displayed by Mintty, pressing Up Arrow and Down Arrow (which normally should navigate through history of previous commands) results in scrolling, pressing 'd' results in the Mintty window being closed.

- I am going to test on another computer as well.

- I am going to test if the other application is also Mintty.

- "sometimes" means "quite often" on this computer (I estimate at least 1 out of 10). The scenario is: I edit some file, then I Alt+Tab into Mintty and run a script, then I Alt+Tab into the other application and edit some more, then I Alt+Tab back into Mintty and run a script etc.

- On this laptop a service called 'LenovoFnAndFunctionKeys' normally runs, but I have long stopped it and switched it to Manual, as it used to display messages on the screen whenever I pressed Caps Lock etc. (https://ibb.co/vmz2GfM).

**Comment by mintty** [comment]
Maybe there's something weird about this "other appliction", so which is it?
Please upload your mintty configuration (.minttyrc file).

**Comment by DoctorNoobingstoneIPresume** [comment]
Thank you.

The other application is ConEmu 191012 (32-bit version) with Far Manager, version 2.0 (build 1807) x86 (32-bit version).

![240623_ConEmu](https://github.com/mintty/mintty/assets/47064117/e4e54b1b-f0de-4cb9-9a82-8175dd288acb)

Mintty is also 32-bit (and, of course, so is my Cygwin installation).

(=> Old version of Cygwin, old version of ConEmu, old version of Far... all on Windows 11 Pro with build of Mintty from recent source code.)

And the Mintty configuration is shown below:

```
$ cat ~/.minttyrc
BoldAsFont=-1
FontHeight=12
Columns=128
Rows=48
ScrollMod=ctrl
ClickTargetMod=alt
AltFnShortcuts=yes
ScrollbackLines=999999
```

**Comment by mintty** [comment]
Well, I've retested with 32-bit mintty, and Windows notepad. Can you please try that as well?
Also, do you press Ctrl anytime during the window switching?
Did you meanwhile try with another computer? Can you try with another keyboard?

**Comment by ivan** [comment]
I also experienced this issue many times on Windows 10 22H2, only after a cygwin upgrade to mintty 3.7.2. I'm pretty sure 3.7.1 was fine.

**Comment by mintty** [comment]
Are you on 32-bit too? What other apps? How often? Any Ctrl key involved?
See questions above; I cannot do anything about this unless I reproduce it.

**Comment by ivan** [comment]
I have 64-bit Windows 10 22H2, 64-bit cygwin and mintty.

I have quit all programs that might intercept the keyboard and managed to reproduce this issue on Windows 10 22H2:

1. Run mintty 3.7.2 as shipped by 64-bit cygwin
2. Press 'c' key and observe normal input processing
3. With mintty still focused, press Ctrl-Alt-Tab and observe the graphical Alt-Tab switcher with thumbnails stay on screen
4. In the Alt-Tab switcher, click on the mintty thumbnail
5. Press 'c' key and observe mintty process it as Ctrl-c
   (It will also continue to be processed as Ctrl-c until you press Ctrl in mintty)

Note that this is not necessarily the only way to reproduce it.

This did not reproduce in mintty 3.7.1.

**Comment by mintty** [comment]
Thank you for this detailed description. I could not yet reproduce it though, but please try the following setting: `OldAltGrDetection=2`.

**Comment by ivan** [comment]
I managed to reproduce this on another Windows 10 22H2 machine (7950X3D desktop instead of my ancient laptop), but of course it might be some obscure configuration I'm not thinking of, e.g. having both English and Japanese keyboards. I could test a clean VM later?

Thanks, I can't reproduce this in mintty 3.7.2 with `OldAltGrDetection=2`.

**Comment by mintty** [comment]
This and another glitch (when using AltGr) could happen if either Ctrl or AltGr was held in auto-repeat. You should be able to tell a difference whether you hit Ctrl+Alt+TAB quickly, or hold Ctrl for another moment, then add Alt+TAB to switch the window.

Uploaded a fix for this problem, I'd appreciate if you test it (download latest from git, install cygwin packages gcc-core and make).

**Comment by mintty** [comment]
Released 3.7.4.

**Comment by ivan** [comment]
Thank you, I think mintty 3.7.4 is working fine now. (Tested without `OldAltGrDetection=2`.)

**Comment by mintty** [comment]
Released 3.7.8.

**Comment by DoctorNoobingstoneIPresume** [comment]
I just wish to say "Thank You for all Your Work !!"... 🤗
